Mathlib/Combinatorics/Additive/AP/Three.lean

```diff
@@ -47,69 +51,72 @@ open Pointwise
 
 variable {F α β 𝕜 E : Type*}
 
-section SalemSpencer
+section ThreeAPFree
 
 open Set
 
 section Monoid
 
 variable [Monoid α] [Monoid β] (s t : Set α)
 
-/-- A multiplicative Salem-Spencer, aka non averaging, set `s` in a monoid is a set such that the
-multiplicative average of any two distinct elements is not in the set. -/
-@[to_additive "A Salem-Spencer, aka non averaging, set `s` in an additive monoid
-is a set such that the average of any two distinct elements is not in the set."]
-def MulSalemSpencer : Prop :=
-  ∀ ⦃a b c⦄, a ∈ s → b ∈ s → c ∈ s → a * b = c * c → a = b
-#align mul_salem_spencer MulSalemSpencer
-#align add_salem_spencer AddSalemSpencer
-
-/-- Whether a given finset is Salem-Spencer is decidable. -/
-@[to_additive "Whether a given finset is Salem-Spencer is decidable."]
-instance {α : Type*} [DecidableEq α] [Monoid α] {s : Finset α} :
-    Decidable (MulSalemSpencer (s : Set α)) :=
+/-- A set is **3GP-free** if it does not contain any non-trivial geometric progression of length
+three.
+
+This is also sometimes called a **non averaging set** or **Salem-Spencer set**. -/
```

Comment:
Is this true of the multiplicative version

Reply:
No, indeed